components/content-service/pkg/storage/minio.go

```diff
@@ -329,11 +335,11 @@ func (rs *DirectMinIOStorage) BackupObject(name string) string {
 }
 
 func (rs *DirectMinIOStorage) bucketName() string {
-	return minioBucketName(rs.Username)
+	return minioBucketName(rs.Username, rs.MinIOConfig.BucketName)
 }
 
 func (rs *DirectMinIOStorage) objectName(name string) string {
-	return minioWorkspaceBackupObjectName(rs.WorkspaceName, name)
+	return minioWorkspaceBackupObjectName(rs.Username, rs.WorkspaceName, name)
```

Comment:
```suggestion
	var username string
	if rs.MinIOConfig.BucketName != "" {
		username = rs.Username
	}
	return minioWorkspaceBackupObjectName(username, rs.WorkspaceName, name)

```
that way if bucket name is set, then we default to new way of doing things, otherwise we maintain backwards compatibility to old way of accessing buckets.